lib/api_local.ml

```diff
@@ -8,50 +8,50 @@ let cache_dir = Caml.Filename.concat cwd "github-api-cache"
 
 (** return the file with a function f applied unless the file is empty;
  empty file:this is needed to simulate 404 returns from github *)
-let get_cache_file_f url f =
+let with_cache_file url f =
   match get_local_file url with
   | Error e ->
     let err_msg = sprintf "error while getting local file: %s\ncached for url: %s" e url in
     Stdio.print_endline err_msg;
-    Lwt.return @@ Error err_msg
-  | Ok "" -> Lwt.return @@ Error "empty file"
-  | Ok file -> Lwt.return @@ Ok (f file)
+    Lwt.return_error err_msg
+  | Ok "" -> Lwt.return_error "empty file"
+  | Ok file -> Lwt.return_ok (f file)
 
 let clean_forward_slashes = String.substr_replace_all ~pattern:"/" ~with_:"_"
 
+(** get a member of the repo cached API call providing
+the member kind (pull, issue, commit, compare, etc),
+_ref (pr number, issue number, commit sha, compare basehead, etc),
+and its Github_j.<kind>_of_string function.
+NB: please save the cache file in the same format *)
+let get_repo_member_cache ~(repo : Github_t.repository) ~kind ~_ref ~of_string =
```

Comment:
```suggestion
let get_repo_member_cache ~(repo : Github_t.repository) ~kind ~ref_ ~of_string =
```

just to be safe, so that a warning is raised when the argument is unused